File: C++/I/05-FORMS/3/form2.cpp
#include "form2.h"
#include "ui_form2.h"

Form2::Form2(QWidget *parent) :
    QDialog(parent),
    ui(new Ui::Form2)
{
    ui->setupUi(this);
    connect(this, SIGNAL(visibleChanged()), this, SLOT(changeVisible()));
}

Form2::~Form2()
{
    delete ui;
}

void Form2::closeEvent(QCloseEvent *)
{
    setVisible(false);
    emit textChanged();
}

void Form2::changeVisible()
{
    setVisible(!isVisible());
    emit textChanged();
    if (isVisible())
        ui->label->setText(QString("Окно открыто в %1-й раз.").arg(++count));
}

File: Python/II/23-TXTEDIT6/1/form2.py
from ui_form2 import *
from PyQt5.QtCore import *
from PyQt5.QtGui import *
from PyQt5.QtWidgets import *

class Form2(Ui_Form2, QDialog):
    def __init__(self, parent):
        super(self.__class__, self).__init__(parent)
        self.setupUi(self)
        self.setFixedSize(self.size())
